index.ts

```diff
@@ -767,6 +767,18 @@ export class EthereumValue {
     return changetype<EthereumTuple>(this.data as u32)
   }
 
+  toTupleArray<T extends EthereumTuple>(): Array<T> {
```

Comment:
Why doesn't this just return `Array<EthereumTuple>`?

Reply:
When there are Tuple type params in the ABI we create new classes for them that extend `EthereumTuple` with accessors for the different properties defined for that Tuple in the ABI. 
We want to return an array of one of those types otherwise we lose the benefit of having the property number and conversion function built into the accessor functions: i.e `MyTuple.propertyName()` would then have to be something like `MyTuple[3].toBytes()`.